pyproject.toml

```diff
@@ -0,0 +1,53 @@
+[tool.poetry]
+name = "up42-py"
+version = "0.32.0"
+description = ""
+authors = ["Your Name <you@example.com>"]
+readme = "README.md"
+
+[tool.poetry.dependencies]
+python = ">=3.8,<3.12"
+requests =  "2.31.0"
+requests-oauthlib = "1.3.1"
+tenacity = "8.2.2"
+tqdm = "4.66.0"
+geojson = "3.0.1"
+geopandas = "0.13.2"
+pystac-client = "0.7.2"
+mkdocs-material = ">=8.5.11"
```

Comment:
let's try to stick to one convention.
Let's either use `">=8.5.11"` or `"^8.5.11"`, but not both :) 

My personal taste would be to use `"^8.5.11"`, since we use this in other python repos as well

Reply:
This can be a good idea, but I thing >= is not the same as ^. I will keep the docs requirements as >= to be consistent with the pip (Hopefully after a couple of iterations we will be removing pip) and then we can deal with the requirements properly. 

The problem is that we have some versions in pip that uses the >= which means any version more than the specified. The ^ operator means that the requirement should be at least the specified version (it means that the version need to exist), and for the docs requirements we need to check which versions are the ones that exist and do not cause incompatibilities. I will add a new ticket to deal with versions in the proper way for poetry, in the mean whiles. This makes the trick to activate poetry and pre-commit which make easier to extend the SDK.

Reply:
yeah, `>=` seems to be better choice.
[https://python-poetry.org/docs/dependency-specification/#caret-requirementssion ](https://python-poetry.org/docs/dependency-specification/#caret-requirementssion)

> The ^ operator means that the requirement should be at least the specified version (it means that the version need to exist)

But that is also true for `>=`, no?